frontend/src/app/core/services/challenge.service.ts

```diff
@@ -42,4 +44,11 @@ export class ChallengeService {
     public updateStar(star: Star): Observable<IChallengePreview> {
         return this.httpService.putRequest<IChallengePreview>(this.challengesRoute, star);
     }
+
+    public postCode(request: CodeRunRequest): Observable<ApiResponse> {
```

Comment:
Can't see the reasoning behind it considering how in other method objects are passed as is it, and well I do not want to create additional copy for storing the data on the other hand if this interface expand in the future it would problematic to have a method definition that spans accross multiple lines of codes (because eslint has constraints on the line length as well)

Reply:
yes, but i would recommend something like this 
ngOnInit() {
  this.initializeSignalR();
  this.loadChallengeAndSetupEditor();
}

Reply:
I want to clarify if we understood each other. I just meant doing like this: 
`public postCode({challengeVersionId, language}: CodeRunRequest) `

Reply:
but inside that function I pass a legitimate object the same object, or you want me to create inside post method a new instance in which I copy the values